src/clear/extensions/openapi.cr

```diff
@@ -0,0 +1,103 @@
+require "../../clear"
+require "openapi-generator"
+
+# The `Serializable` module automatically generates an OpenAPI Operations representation of the class or struct when extended.
+#
+# ### Example
+#
+# ```
+# class ClearModelExample
+#   include Clear::Model
+#   extend OpenAPI::Generator::Serializable
+
+#   column id : Int64, primary: true, mass_assign: false, example: "123"
+#   column email : String, mass_assign: true, example: "default@gmail.com"
+# end
+# # => {
+# #     "required": [
+# #       "id",
+# #       "email"
+# #     ],
+# #     "type": "object",
+# #     "properties": {
+# #       "id": {
+# #         "type": "integer",
+# #         "readOnly": true,
+# #         "example": "123"
+# #       },
+# #       "email": {
+# #         "type": "string",
+# #         "writeOnly": true,
+# #         "example": "default@gmail.com"
+# #       }
+# #     }
+# #   }
+# ```
+#
+# ### Usage
+#
+# Extending this module adds a `self.to_openapi_schema` that returns an OpenAPI representation
+# inferred from the shape of the class or struct.
+#
+# The class name is also registered as a global [component schema](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#components-object)
+# and will be available for referencing from any `Controller` annotation from a [reference object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#referenceObject).
+#
+# **See:** `OpenAPI::Generator::Controller::Schema.ref`
+#
+# NOTE: **Calling `to_openapi_schema` programatically is unnecessary.
+# The `Generator` will take care of serialization while producing the openapi yaml file.**
+module OpenAPI::Generator::Serializable
```

Comment:
Is it necessary to reopen this module?

Reply:
It's a clean and straightforward way to do this. It is not the only way, but otherwise I would need to do some macro sudoku to access the `OpenAPI::Generator::Serializable.SERIALIZABLE_CLASSES` [constant delayed at compile-time](https://github.com/place-labs/openapi-generator/blob/74c0446dfd28267143b5526fc0731b60b9ccb1c2/src/openapi-generator/serializable.cr#L289), some more abstractions and detailed specs. So I figure this is fine for now. If we want to make the abstraction cleaner, we can do that at a later date.